### PR TITLE
Fix various typos

### PR DESCRIPTION
**Pull request by bdrung**
Lintian found various typos. The first commit fixes the trivial typos. The second commit fixes the typo in the heat module/state parameter. The rename is documented and support for the misspelled version for backward compatibility is added.

### CODE REVIEW

File: salt/modules/heat.py [hunk 1]
```diff
@@ -496,11 +496,19 @@ def create_stack(name=None, template_file=None, enviroment=None,
...
 
+    .. versionadded:: 2017.7.5,2018.3.1
+
+        The spelling mistake in parameter `enviroment` was corrected to `environment`.
+        The misspelled version is still supported for backward compatibility.
+
     '''
+    if environment is None:
+        # Support for misspelled parameter
+        environment = enviroment
```
Comment (rallytime) [inline salt/modules/heat.py]: I think we should put the misspelled version on a deprecation path. Look for the `warn_util` function of how to do this and we can remove it in 2 feature releases.

File: salt/modules/heat.py [hunk 2]
```diff
@@ -681,11 +689,19 @@ def update_stack(name=None, template_file=None, enviroment=None,
...
```
Comment (rallytime) [inline salt/modules/heat.py]: Same comment here. :D

File: salt/states/heat.py [hunk 3]
```diff
@@ -152,10 +157,18 @@ def deployed(name, template=None, enviroment=None, params=None, poll=5,
...
 
+    .. versionadded:: 2017.7.5,2018.3.1
+
+        The spelling mistake in parameter `enviroment` was corrected to `environment`.
+        The misspelled version is still supported for backward compatibility.
+
     '''
+    if environment is None and 'enviroment' in connection_args:
+        # Support for misspelled parameter
+        environment = connection_args.pop('enviroment')
```
Comment (rallytime) [inline salt/states/heat.py]: The `warn_util` deprecation notice should happen here as well.

**Review by rallytime — CHANGES_REQUESTED** [review]
Thanks @bdrung! It's always nice to get these cleaned up.

I have one request, but once that is changed, this looks good to me.

**Comment by bdrung** [comment]
Took me some seconds to figure out that you meant `warn_until` instead of `warn_util`. :smile: Implemented deprecation warning as requested.

**Comment by rallytime** [comment]
Haha woops! Speaking of typos.... 😁 

This looks great now!

**Review by rallytime — APPROVED** [review]

**Comment by bdrung** [comment]
Look at the diff and you will find typo fixes for commit messages that fix typos. :laughing:

**Review by terminalmage — APPROVED** [review]

**Pull request merged**